testsuite/testcases/src/state_sync_performance.rs

```diff
@@ -23,81 +21,86 @@ impl Test for StateSyncPerformance {
-        if validator_synced_version == 0.0 {
+        // Fetch the highest synced version from the swarm
+        let highest_synced_version =
+            runtime.block_on(async { ctx.swarm().get_highest_synced_version().await.unwrap_or(0) });
+        if highest_synced_version == 0 {
             return Err(anyhow::format_err!(
-                "Validator synced zero transactions! Something has gone wrong!"
+                "The swarm has synced 0 versions! Something has gone wrong!"
             ));
         }
-        println!(
-            "The validator is now synced at version: {}",
-            validator_synced_version
+        info!(
+            "Syncing to target version at : {:?}",
+            highest_synced_version
         );
 
-        // 4. restart the fullnode so that it starts state syncing to catch up
-        let fullnode = ctx.swarm().full_node_mut(*fullnode_id).unwrap();
-        // do data cleanup
-        fullnode.clear_storage()?;
-        println!("The fullnode is going to restart");
-        runtime.block_on(fullnode.start())?;
-        println!(
-            "The fullnode is now up. Waiting for it to state sync to the expected version: {}",
-            validator_synced_version
-        );
-        let start_instant = Instant::now();
-        let fullnode_metric_port = fullnode.expose_metric()?;
-        while fullnode
-            .counter(STATE_SYNC_COMMITTED_COUNTER_NAME, fullnode_metric_port)
-            .unwrap_or(0.0)
-            < validator_synced_version
-        {
-            thread::sleep(Duration::from_secs(1));
+        // Restart the fullnodes so they start syncing from a fresh state
+        for fullnode_id in &all_fullnodes {
+            let fullnode = ctx.swarm().full_node_mut(*fullnode_id).unwrap();
+            runtime.block_on(async { fullnode.start().await })?;
         }
-        println!(
-            "The fullnode has caught up to version: {}",
-            validator_synced_version
-        );
+        info!("Restarting all the fullnodes and waiting for them to catchup.");
+
+        // Wait for all fullnodes to catch up to the highest synced version
+        let timer = Instant::now();
+        runtime.block_on(async {
+            ctx.swarm()
+                .wait_for_all_nodes_to_catchup(
+                    Instant::now() + Duration::from_secs(MAX_FULLNODE_SYNC_SECS),
+                )
+                .await
+        })?;
+        let time_to_state_sync = timer.elapsed().as_secs();
 
         // Calculate the state sync throughput
-        let time_to_state_sync = start_instant.elapsed().as_secs();
         if time_to_state_sync == 0 {
             return Err(anyhow::format_err!(
                 "The time taken to state sync was 0 seconds! Something has gone wrong!"
             ));
         }
-        let state_sync_throughput = validator_synced_version as u64 / time_to_state_sync;
+        let state_sync_throughput = highest_synced_version as u64 / time_to_state_sync;
```

Comment:
How about we add success criteria for state sync throughput as well? That way we can start measuring it and also fail in case the throughput dropped below a threshold?

Reply:
Of course, you can configure different throughput for local mode vs K8, which we do for TPS tests as well.

Reply:
Done. I've hacked it it using the existing `check_success` functionality. But, I've added a comment that we might want to make it more generic outside of just a `txn_stats` and `duration` interface.